blocks/bricks/sequence_generators.py

```diff
@@ -136,6 +137,34 @@ def _push_allocation_config(self):
 
     @application
     def cost(self, application_call, outputs, mask=None, **kwargs):
+        """Returns sum of generation costs for output sequences.
+
+        Parameters
+        ----------
+        outputs : :class:`~tensor.TensorVariable`
+            The 3(2) dimensional tensor containing output sequences.
+            The dimension 0 must stand for time, the dimension 1 for the
+            position on the batch.
+        mask : :class:`~tensor.TensorVariable`
+            The binary matrix identifying fake outputs.
+
+        Notes
+        -----
+        The contexts are expected as keyword arguments.
+
+        """
+        # Compute the sum of costs
+        cost = self.cost_matrix(outputs, mask=mask, **kwargs).sum()
```

Comment:
Same argument holds for that. If your batches are of different sizes, dividing by batch size will give greater weight to smaller batches. That can be problematic if you are e.g. forcing batches to only contain sequences of equal length. Outliers (with potentially only one example per batch) will weigh very heavily then.